### PR TITLE
[Model Monitoring] Support running applications locally with `app.evaluate()`

### PR DESCRIPTION
**Pull request by jond01**
Add the `evaluate` method to `ModelMonitoringApplicationBase` class, currently without parameters/inputs.
Implements [ML-8357](https://iguazio.atlassian.net/browse/ML-8357).

[ML-8357]: https://iguazio.atlassian.net/browse/ML-8357?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/model_monitoring/test_applications/test_base.py [hunk 1]
```diff
@@ -14,15 +14,56 @@
...
+class NoOpApp(ModelMonitoringApplicationBase):
+    def do_tracking(self, monitoring_context: MonitoringApplicationContext):
+        pass
+
+
+class InProgressApp0(ModelMonitoringApplicationBase):
+    def do_tracking(
+        self, monitoring_context: MonitoringApplicationContext
+    ) -> ModelMonitoringApplicationResult:
+        monitoring_context.logger.info("Ok let's try")
```
Comment (assaf758) [inline tests/model_monitoring/test_applications/test_base.py:35]: Better change logged string to something like "This test-app is failing on purpose - ignore the failure!"

File: tests/model_monitoring/test_applications/test_base.py [hunk 2]
```diff
@@ -14,15 +14,56 @@
...
+    @pytest.fixture(autouse=True)
+    def _set_project(cls) -> None:
+        mlrun.get_or_create_project("test")
 
-    App()
+    @staticmethod
+    def test_local_no_params() -> None:
+        func_name = "test-app"
+        run = InProgressApp0.evaluate(func_path=__file__, func_name=func_name)
+        assert run.state() == "created"  # Should be "error"
```
Comment (assaf758) [inline tests/model_monitoring/test_applications/test_base.py:67]: I think you raised a bug on this? please add reference to it

**Review by assaf758 — APPROVED** [review]

File: tests/model_monitoring/test_applications/test_base.py (hunk 1 above)
Comment (jond01) [reply tests/model_monitoring/test_applications/test_base.py:35]: Done in #6825

File: tests/model_monitoring/test_applications/test_base.py (hunk 2 above)
Comment (jond01) [reply tests/model_monitoring/test_applications/test_base.py:67]: Done in #6825

**Pull request merged**